src/react-app.js

```diff
@@ -386,56 +410,22 @@ function renderReactApp({
                     />
                     Final
                   </Tab>
-                  {extraCredit.length ? (
-                    <Tab css={{display: 'flex', alignItems: 'center'}}>
-                      <RiTrophyLine
-                        size="18"
-                        color={theme.textLightest}
-                        css={{marginRight: 5}}
-                      />
-                      <span>Extra Credit</span>
-                    </Tab>
-                  ) : null}
                 </TabList>
                 <TabPanels>
                   <TabPanel>
-                    <Sandbox
+                    <FileTabs
+                      key={exerciseNumber}
                       isOpen={tabIndex === 0}
-                      isolatedPath={exercise.isolatedPath}
-                      isolatedPathLinkContent="Open exercise on isolated page"
-                    >
-                      {renderedTabs.current.has(0) ? (
-                        <iframe
-                          title="Exercise"
-                          src={exercise.isolatedPath}
-                          css={{border: 'none', width: '100%', height: '100%'}}
-                        />
-                      ) : null}
-                    </Sandbox>
+                      files={exercise}
+                    />
                   </TabPanel>
                   <TabPanel>
-                    <Sandbox
+                    <FileTabs
+                      key={exerciseNumber}
```

Comment:
I guess I forgot to explain the purpose of these `key`s - if not obvious, they replace this deleted workaround:

```js
    // handle this case:
    // 1. Select extra credit tab
    // 2. navigate to exercise with no extra credit
    // we want to set the tab index to 0 so they're not looking at nothing.
    React.useEffect(() => {
      if (!extraCredit.length && tabIndex === 2) setTabIndex(0)
    }, [extraCredit.length, tabIndex])
```

If you think someone might be tempted to remove them as "not needed", please reintroduce the explanation comment ;)

Reply:
I'll keep that in mind. Thanks!